src/flint/types/fmpq_mpoly.pyx

```diff
@@ -0,0 +1,600 @@
+                raise ValueError("Cannot automatically coerce contexts")
+            init_fmpq_mpoly(self, ctx)
+            fmpz_mpoly_set(self.val.zpoly, (<fmpz_mpoly>val).val, (<fmpz_mpoly> val).ctx.val)
+            fmpq_one(self.val.content)
+            fmpq_mpoly_reduce(self.val, self.ctx.val)
+        elif isinstance(val, dict):
+            if ctx is None:
+                if len(val) == 0:
+                    raise ValueError("Need context for zero polynomial")
+                k = list(val.keys())[0]
+                if not isinstance(k, tuple):
+                    raise ValueError("Dict should be keyed with tuples of integers")
+                ctx = fmpq_mpoly_ctx.get_context(len(k))
+            x = ctx.fmpq_mpoly_from_dict(val)
+            #XXX this copy is silly, have a ctx function that assigns an fmpz_mpoly_t
+            init_fmpq_mpoly(self, ctx)
+            fmpq_mpoly_set(self.val, (<fmpq_mpoly>x).val, self.ctx.val)
+        elif isinstance(val, str):
+            if ctx is None:
+                raise ValueError("Cannot parse a polynomial without context")
+            val = bytes(val, 'utf-8')
+            init_fmpq_mpoly(self, ctx)
+            fmpq_mpoly_set_str_pretty(self.val, val, self.ctx.c_names, self.ctx.val)
+            fmpq_mpoly_sort_terms(self.val, self.ctx.val)
+        else:
+            v = any_as_fmpq(val)
+            if v is NotImplemented:
+                raise TypeError("cannot create fmpz_mpoly from type %s" % type(val))
+            if ctx is None:
+                raise ValueError("Need context to convert  fmpz to fmpq_mpoly")
+            init_fmpq_mpoly(self, ctx)
+            fmpq_mpoly_set_fmpq(self.val, (<fmpq>v).val, self.ctx.val)
+
+    def __nonzero__(self):
+        return not fmpq_mpoly_is_zero(self.val, self.ctx.val)
+
+    def __bool__(self):
+        return not fmpq_mpoly_is_zero(self.val, self.ctx.val)
+
+    def is_one(self):
+        return fmpq_mpoly_is_one(self.val, self.ctx.val)
+
+    def __richcmp__(self, other, int op):
+        if op != 2 and op != 3:
+            return NotImplemented
+        if typecheck(self, fmpq_mpoly) and typecheck(other, fmpq_mpoly):
+            if (<fmpq_mpoly>self).ctx is (<fmpq_mpoly>other).ctx:
+                if op == 2:
+                    return bool(fmpq_mpoly_equal((<fmpq_mpoly>self).val, (<fmpq_mpoly>other).val, (<fmpq_mpoly>self).ctx.val))
+                else:
+                    return not bool(fmpq_mpoly_equal((<fmpq_mpoly>self).val, (<fmpq_mpoly>other).val, (<fmpq_mpoly>self).ctx.val))
+            else:
+                if op == 2:
+                    return False
+                else:
+                    return True
+        if op == 2:
+            return not bool(self - other)
+        else:
+            return bool(self - other)
+
+    def context(self):
+        return self.ctx
+
+    def __len__(self):
+        return fmpq_mpoly_length(self.val, self.ctx.val)
```

Comment:
I just went to the Flint docs to check what this does. It would be good to have docstrings that show what each function does.